spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/NumFiles.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.spark.source;
+
+import java.text.NumberFormat;
+import org.apache.spark.sql.connector.metric.CustomMetric;
+
+public class NumFiles implements CustomMetric {
+
+  @Override
+  public String name() {
+    return "numFiles";
+  }
+
+  @Override
+  public String description() {
+    return "number of files read";
+  }
+
+  @Override
+  public String aggregateTaskMetrics(long[] taskMetrics) {
+    long sum = initialValue;
+    for (int i = 0; i < taskMetrics.length; i++) {
+      sum += taskMetrics[i];
+    }
+    return NumberFormat.getIntegerInstance().format(sum);
+  }
```

Comment:
[question] should we make this return "total(min, max, med)" we already have all the metrics required, this should give us even deeper insights on files distribution at task level. WDYT ?

Reply:
@singhpk234 thank you for your interesting suggesions!
I am open to presenting  the min, median and max as well, but I wonder how useful those statistics really are. My impression is that Iceberg scan planning works well enough that the number of files to scan are usually very evenly distributed (except for cases like where there are two `InputPartition`s). Usually skew occurs with times (which can of course show great variation), or with number of records after a shuffle (rather than a `BatchScan`).
@rdblue, @aokolnychyi, @RussellSpitzer what do you think?

Reply:
I though the way these are shown in the UI are per task as well? Or am I remembering that wrong. IE we should have all these values (but not the calculated sum?) If not I think that would be good to report, I would suggest probably

min, 25percentile, median, 75percentile, max?

Reply:
@RussellSpitzer I think you may be thinking of task metrics that Spark itself collects (by default, Duration, GC Time,  Input Size / Records, and Shuffle Size /Records are shown), and are shown with the min, 25th percentile, median, 75th percentile, and max.
For custom `SQLMetric`s, one implements the `aggregateTaskMetrics` and we'd need to do the calculation ourselves if we want such statistics. Spark doesn't do it for us. Spark sends all the values it collected from all the `PartitionReader`s for a given custom metric in a `long[]` to this `aggregateTaskMetrics`; that's all. It also doesn't separately calculate these statistics either.